common/testing/taskpoller/taskpoller.go

```diff
@@ -0,0 +1,284 @@
+package taskpoller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	failurepb "go.temporal.io/api/failure/v1"
+	historypb "go.temporal.io/api/history/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/debug"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/rpc"
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/service/history/consts"
+)
+
+type (
+	TaskPoller struct {
+		t          require.TestingT
+		_logger    log.Logger
+		_client    workflowservice.WorkflowServiceClient
+		_namespace string
```

Comment:
This is odd, I know, but I was getting compiler errors for `client` etc. as it clashes with the FunctionalSuite it is embedded in. I can also wrap these in an inner struct.

Reply:
> Why do you want to embed it? 

Less typing ;)